arcgis-ios-sdk-samples/Layers/Export tiles/ExportTilesViewController.swift

```diff
@@ -17,154 +17,233 @@ import UIKit
-            for file in files {
-                if file.hasSuffix(".tpk") {
-                    let url = documentDirectoryURL.appendingPathComponent(file)
-                    try FileManager.default.removeItem(at: url)
-                }
+    /// Get destination URL for the tile package.
+    private func getDownloadURL(fileFormat: TilePackageFormat) -> URL {
+        // If the downloadFileURL ends with ".tpk", the tile cache will use
+        // the legacy compact format. If the downloadFileURL ends with ".tpkx",
+        // the tile cache will use the current compact version 2 format.
+        // See more in the doc of
+        // `AGSExportTileCacheTask.exportTileCacheJob(with:downloadFileURL:)`.
+        temporaryFolderURL.appendingPathComponent("myTileCache.\(fileFormat.fileExtension)")
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func exportTilesBarButtonTapped(_ sender: UIBarButtonItem) {
+        if let exportJob = job {
+            // If it is downloading, cancel download.
+            exportJob.progress.cancel()
+            job = nil
+        } else {
+            // Otherwise, try to download when exporting tiles is allowed.
+            guard let mapServiceInfo = exportTask.mapServiceInfo, mapServiceInfo.exportTilesAllowed else { return }
+            // Create an action sheet to choose export format.
+            let alertController = UIAlertController(
+                title: "Choose Export Format",
+                message: nil,
+                preferredStyle: .actionSheet
+            )
+            let allowedFormats: [TilePackageFormat] = mapServiceInfo.exportTileCacheCompactV2Allowed ? [.tpk, .tpkx] : [.tpk]
+            allowedFormats.forEach { fileFormat in
+                alertController.addAction(
+                    UIAlertAction(title: fileFormat.description, style: .default) { [unowned self] _ in
+                        self.initiateDownload(exportTask: exportTask, downloadFileURL: self.getDownloadURL(fileFormat: fileFormat))
+                    }
+                )
+            }
+            let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+            alertController.addAction(cancelAction)
+            alertController.popoverPresentationController?.barButtonItem = sender
+            present(alertController, animated: true)
+        }
+    }
+    
+    @IBAction func closeButtonTapped(_ sender: UIButton) {
+        // Hide the preview and background.
+        visualEffectView.isHidden = true
+        // Release the map in order to free the tiled layer.
+        previewMapView.map = nil
+        // Remove all files in the sample-specific temporary folder.
+        guard let files = try? FileManager.default.contentsOfDirectory(at: temporaryFolderURL, includingPropertiesForKeys: nil), !files.isEmpty else { return }
+        files.forEach { filePath in
+            try? FileManager.default.removeItem(at: filePath)
+        }
```

Comment:
Could the temporary directory just be removed altogether?

```suggestion
        do {
            try FileManager.default.removeItem(at: temporaryFolderURL)
        } catch {
            print("Error removing temporary directory: \(error.localizedDescription)")
        }
```

Reply:
Mentioned above in https://github.com/Esri/arcgis-runtime-samples-ios/pull/986#discussion_r512336655

If the temp folder is removed before a user quit the sample, there would possibly be a case that the user want to export another tile package, while the temp folder does not exist, which leads to an error.

Reply:
Could the temp directory (not the URL, but the directory itself) be created in `makeDownloadURL(fileFormat:)`? That way if it is deleted, it will be re-created when needed, right?

Reply:
Great idea! In this way I can also remove the closure to init the temp directory URL.